src/widgets/send-message.jsx

```diff
@@ -233,6 +243,13 @@ class SendMessage extends React.PureComponent {
         {this.state.quote}
       </div>) : null;
     const audioEnabled = this.state.audioAvailable && this.props.onAttachAudio;
+    const inputField = (<textarea id="sendMessage" placeholder={prompt}
+      value={this.state.message} onChange={this.handleMessageTyping}
+      onKeyPress={this.handleKeyPress}
+      ref={(ref) => {this.messageEditArea = ref;}}
+      autoFocus />);
+    const inputArea = this.props.onAttachFile ?
+      (<DragAndDrop actionPrompt={formatMessage(messages.icon_title_attach_file)} onDrop={this.handleDropAttach}>{inputField}</DragAndDrop>) : {inputField};
```

Comment:
Why do you want to wrap inputField into `<DragAndDrop>`? Why not show DnD above the input as a div?

Reply:
There is nothing wrong with using the z-index. It's just has to be carefully selected not to cover other elements, like context menu.
Otherwise the top element is the last child of the parent. Make the element `position: absolute` and make it the last element.

Anyway, this code is fine. You can commit it as is, I will tweak the UI a bit before release. Thanks.